packages/utils/src/worldwide.ts

```diff
@@ -55,6 +55,13 @@ export interface InternalGlobal {
       [key: string]: Function;
     };
   };
+  /**
+   * Raw module metadata that is injected by bundler plugins.
+   *
+   * Keys are `error.stack` strings, values are metadata objects.
+   * Once the stack has been parsed, the value is set to `false` so they do not get parsed multiple times.
+   */
+  __MODULE_METADATA__?: Record<string, object | false>;
```

Comment:
> let's just have a cache (Map) we can use to look up stacktrace -> metadata

We are not looking up stacktrace -> metadata, we are looking up filename -> metadata. The stacktraces are only used as a reliable way to get the first `filename` in the stack. As chunks are loaded (potentially much later when navigating an SPA), `__MODULE_METADATA__` will have additions and we should ensure that we don't parse any of these more than once since there may be 100+ with a large app with many remote chunks.

The resulting cache after parsing has `filename` keys and metadata values so we can quickly lookup metadata for a `filename`. However, there is no way with only these to know if we've already parsed a specific stacktrace. 

Rather than set this to `false`, we could alternatively store a `Set` of previously parsed `error.stack`s but this would use extra memory.

Reply:
Ok I just realized we're not caching the stack traces in the implementation for debug IDs so let's not reuse that logic :D  . I need to fix that.

My suggestion for this PR: Let's keep `__MODULE_METADATA__` (or whatever we're gonna end up calling it) immutable. This lets us rely on it in any internal implementation details across updates in the plugin and SDK.

However, we're caching parsed stack traces and the metadata doesn't really matter I think. I would just do it on some separate object/map.